### PR TITLE
mgr: turn on balancer in upmap mode by default

### PR DESCRIPTION
**Pull request by neha-ojha**
This PR turns the balancer on by default in the upmap mode. Since the upmap mode requires require_min_compat_client luminous, we are also changing the default value of `mon_osd_initial_require_min_compat_client` from jewel to luminous. It is still possible to turn the balancer off.

In earlier versions, the balancer was included in the always_on_modules list, but needed to be turned on explicitly using the `ceph balancer on` command. 

Signed-off-by: Neha Ojha <nojha@redhat.com>

### CODE REVIEW

**Comment by neha-ojha** [comment]
No related failures: http://pulpito.ceph.com/nojha-2020-04-12_18:19:20-rados-wip-balancer-on-2-distro-basic-smithi/

File: qa/standalone/mon/osd-pool-df.sh [hunk 1]
```diff
@@ -58,10 +58,11 @@ function TEST_ceph_df() {
     local ec_poolname=testcephdf_erasurecode
     create_pool $rep_poolname 6 6 replicated
     create_pool $ec_poolname 6 6 erasure ec42profile
+    sleep 2
```
Comment (jdurgin) [inline qa/standalone/mon/osd-pool-df.sh:61]: sleeps make me think the test may race - is there a way to wait for the desired state instead?

**Comment by neha-ojha** [comment]
@jdurgin this is ready for final review

File: PendingReleaseNotes [hunk 2]
```diff
@@ -324,3 +324,10 @@
 
 * librbd now inherits the stripe unit and count from its parent image upon creation.
   This can be overridden by specifying different stripe settings during clone creation.
+
+* The balancer on by default in the upmap mode. Since the upmap mode requires
+  ``require_min_compat_client`` luminous, we are also changing the default value
```
Comment (jdurgin) [inline PendingReleaseNotes:329]:
```suggestion
  ``require_min_compat_client`` luminous, new clusters will only support luminous and newer clients by default. Existing clusters can enable upmap support by running ``ceph osd set-require-min-compat-client luminous``.
```

File: PendingReleaseNotes [hunk 3]
```diff
@@ -324,3 +324,10 @@
...
+  of ``mon_osd_initial_require_min_compat_client`` from jewel to luminous. It is
```
Comment (jdurgin) [inline PendingReleaseNotes:330]: mon_osd_initial... isn't documented, better to explain its effect than to spell it out here

File: doc/rados/operations/balancer.rst [hunk 4]
```diff
@@ -19,17 +19,17 @@ The current status of the balancer can be checked at any time with::
 Automatic balancing
 -------------------
 
-The automatic balancing can be enabled, using the default settings, with::
-
-  ceph balancer on
-
-The balancer can be turned back off again with::
+The automatic balancing feature is enabled by default in the ``upmap``
+mode. The balancer can be turned back off with::
```
Comment (jdurgin) [inline doc/rados/operations/balancer.rst:23]:
```suggestion
mode. The balancer can be turned off with::
```

File: doc/rados/operations/balancer.rst [hunk 5]
```diff
@@ -19,17 +19,17 @@ The current status of the balancer can be checked at any time with::
...
 
   ceph balancer off
 
-This will use the ``crush-compat`` mode, which is backward compatible
-with older clients, and will make small changes to the data
-distribution over time to ensure that OSDs are equally utilized.
+The ``upmap`` mode needs ``require_min_compat_client`` luminous, which
```
Comment (jdurgin) [inline doc/rados/operations/balancer.rst:27]: you can reference rados/operations/upmap.rst instead - the underscores vs dashes in the user command may confuse users

File: doc/rados/operations/upmap.rst [hunk 6]
```diff
@@ -12,21 +12,15 @@ clients understand the new *pg-upmap* structure in the OSDMap.
 Enabling
 --------
 
-To allow use of the feature, you must tell the cluster that it only
-needs to support luminous (and newer) clients with::
+This module is on by default. The cluster must only have luminous (and newer)
+clients. You can the turn the balancer off with::
 
-  ceph osd set-require-min-compat-client luminous
```
Comment (jdurgin) [inline doc/rados/operations/upmap.rst:18]: this part is still relevant for upgraded clusters, I'd leave it - it's also the only documentation of this command

**Comment by neha-ojha** [comment]
jenkins render docs

**Comment by neha-ojha** [comment]
jenkins render docs

**Pull request merged**